### PR TITLE
fix: add mac catalyst specific workarounds

### PR DESCRIPTION
**Pull request by gnattu**
The Mac Catalyst target has some behavioral differences due to UIKit being replaced by AppKit in some instances. Additionally, it has different conventions for controls, such as the volume control. On iOS, the volume control is expected to be system-wide, but on macOS, it is expected to be app-specific. To match the iOS App's appearance, the titlebar also needs to be hidden, but this cannot be done with SwiftUI and has to use UIKit methods which is weird. 

This PR addressed most outstanding issues I found during my usage, but there could be more. If you found more feel free to tell me and I'm going to look at them.

### CODE REVIEW

File: Multiplatform/Account/AccountSheet.swift
Comment (rasmuslos) [inline Multiplatform/Account/AccountSheet.swift:1]: You can just use the environment [`dismiss`](https://developer.apple.com/documentation/swiftui/environmentvalues/dismiss/) function

Comment (gnattu) [reply Multiplatform/Account/AccountSheet.swift:1]: Oh, speaking about this, we still have some code using the deprecated `presentationMode.wrappedValue.dismiss`, do you want me to correct these in this PR as well or you want to do it yourself?

Comment (rasmuslos) [reply Multiplatform/Account/AccountSheet.swift:1]: Would be great if you could do it 👍

**Comment by gnattu** [comment]
https://github.com/rasmuslos/AmpFin/blob/e256426604173b2c875cbb151e3e2cdef34ff61e/AmpFinKit/Sources/AFPlayback/LocalAudioEndpoint/LocalAudioEndpoint%2BHelper.swift#L71

Also, during debugging I found this line keep querying the system volume periodically but this code will introduce inter-process communication as the system volume can only be fetched this way, and such fetching could happen in the main thread which will potentially introduce unwanted delay and Xcode warned me about this. Can we just use the cached in-process `volume` inside the `LocalAudioEndpoint` instead?

**Comment by rasmuslos** [comment]
Thanks!

**Pull request merged**